poetry/installation/pip_installer.py

```diff
@@ -143,7 +143,7 @@ def requirement(self, package, formatted=False):
 
         if package.source_type in ["file", "directory"]:
             if package.root_dir:
-                req = os.path.join(package.root_dir, package.source_url)
+                req = os.path.join(package.root_dir.as_posix(), package.source_url)
```

Comment:
```suggestion
                req = package.root_dir / package.source_url
```

Reply:
Thanks for the suggestion and +1.
I added a commit to avoid using `os.path.join` but still convert the result to `str` as `req = (package.root_dir / package.source_url).as_posix()` because I think `req` is expected to be `str`.